File: FitBook.Model/SearchObjects/TrainingTermSearchObject.cs
using FitBook.Model.Enums;

namespace FitBook.Model.SearchObjects;

public class TrainingTermSearchObject : BaseSearchObject
{
    public int? TrainingId { get; set; }
    public int? TrainerId { get; set; }
    public int? HallId { get; set; }
    public TrainingTermStatus? Status { get; set; }
    public DateTime? StartFromUtc { get; set; }
    public DateTime? StartToUtc { get; set; }
    public bool? IsActive { get; set; }
}

File: UI/fitbook_desktop/lib/models/search_objects/training_term_search_object.dart
import '../common/base_search_object.dart';
import '../enums/training_term_status.dart';

/// Mirrors `FitBook.Model.SearchObjects.TrainingTermSearchObject`.
class TrainingTermSearchObject extends BaseSearchObject {
  const TrainingTermSearchObject({
    super.page,
    super.pageSize,
    super.search,
    super.includeTotalCount,
    this.trainingId,
    this.trainerId,
    this.hallId,
    this.status,
    this.startFromUtc,
    this.startToUtc,
    this.isActive,
  });

  final int? trainingId;
  final int? trainerId;
  final int? hallId;
  final TrainingTermStatus? status;
  final DateTime? startFromUtc;
  final DateTime? startToUtc;
  final bool? isActive;

  @override
  Map<String, dynamic> toQueryParameters() {
    final params = super.toQueryParameters();
    if (trainingId != null) params['trainingId'] = trainingId;
    if (trainerId != null) params['trainerId'] = trainerId;
    if (hallId != null) params['hallId'] = hallId;
    if (status != null) params['status'] = status!.value;
    if (startFromUtc != null) params['startFromUtc'] = startFromUtc!.toIso8601String();
    if (startToUtc != null) params['startToUtc'] = startToUtc!.toIso8601String();
    if (isActive != null) params['isActive'] = isActive;
    return params;
  }
}
